Applications/spire/source/charting/chart_view.cpp

```diff
@@ -248,6 +265,26 @@ void ChartView::showEvent(QShowEvent* event) {
   QWidget::showEvent(event);
 }
 
+void ChartView::update_auto_scale() {
+  if(m_candlesticks.size() > 1) {
+    auto auto_scale_top = m_candlesticks[1].GetHigh();
```

Comment:
What if there's only one candlestick?

Reply:
I updated the if statement above it to check for that.

I noticed that in the old chart that auto-scale wouldn't do anything if there is only 1 candlestick completely visible.